gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java

```diff
@@ -267,7 +276,128 @@ public Spec getSpec(URI specUri, String version) throws IOException, SpecNotFoun
     return Optional.of(this.specStoreURI);
   }
 
-  protected void setPreparedStatement(PreparedStatement statement, Spec spec, String tagValue) throws SQLException {
+  static String createGetPreparedStatement(FlowSpecSearchObject flowSpecSearchObject, String tableName)
+      throws IOException {
+    String baseStatement = String.format(GET_STATEMENT, tableName);
+    List<String> conditions = new ArrayList<>();
+
+    if (flowSpecSearchObject.getFlowSpecUri() != null) {
+      conditions.add("spec_uri = ?");
+    }
+
+    if (flowSpecSearchObject.getFlowGroup() != null) {
+      conditions.add("flow_group = ?");
+    }
+
+    if (flowSpecSearchObject.getFlowName() != null) {
+      conditions.add("flow_name = ?");
+    }
+
+    if (flowSpecSearchObject.getTemplateURI() != null) {
+      conditions.add("template_uri = ?");
+    }
+
+    if (flowSpecSearchObject.getUserToProxy() != null) {
+      conditions.add("user_to_proxy = ?");
+    }
+
+    if (flowSpecSearchObject.getSourceIdentifier() != null) {
+      conditions.add("source_identifier = ?");
+    }
+
+    if (flowSpecSearchObject.getDestinationIdentifier() != null) {
+      conditions.add("destination_identifier = ?");
+    }
+
+    if (flowSpecSearchObject.getSchedule() != null) {
+      conditions.add("schedule = ?");
+    }
+
+    if (flowSpecSearchObject.getSchedule() != null) {
+      conditions.add("modified_time = ?");
+    }
```

Comment:
There's two conditions added here for schedule and only one gets set in `setGetPreparedStatement`, is this a mistake?